src/codegen/translate_to_glib.rs

```diff
@@ -12,9 +12,10 @@ impl TranslateToGlib for chunk::parameter_ffi_call_in::Parameter {
         use analysis::conversion_type::ConversionType::*;
         match ConversionType::of(library, self.typ) {
             Direct => self.name.clone(),
-            Scalar => format!("{}{}", self.name, ".to_glib()"),
+            Scalar => format!("{}{}{}",
+                              self.name, if !*self.nullable { "" } else { ".into()" }, ".to_glib()"),
             Pointer => {
-                let (left, right) = to_glib_xxx(self.transfer, self.ref_mode);
+                let (left, right) = to_glib_xxx(self.transfer, self.ref_mode, self.is_into);
```

Comment:
Hum... We can't do it on every parameter so we still need to check the enum variant. I think it's good as is (but maybe I misunderstood you, if so, can you explain me what you have in mind more precisely please?).